src/baldr/graphreader.cc

```diff
@@ -630,5 +645,43 @@ std::unordered_set<GraphId> GraphReader::GetTileSet(const uint8_t level) const {
   return tiles;
 }
 
+AABB2<PointLL> GraphReader::GetMinimumBoundingBox(const AABB2<PointLL>& bb) {
+  // Iterate through all the tiles that intersect this bounding box
+  const auto& ids = TileHierarchy::GetGraphIds(bb);
+  AABB2<PointLL> min_bb{PointLL{}, PointLL{}};
+  for (const auto& tile_id : ids) {
+    // Don't take too much ram
+    if (OverCommitted())
+      Clear();
+
+    // Look at every node in the tile
+    const auto* tile = GetGraphTile(tile_id);
+    for (uint32_t i = 0; tile && i < tile->header()->nodecount(); i++) {
+
+      // If the node is within the input bounding box
+      const auto* node = tile->node(i);
+      auto node_ll = node->latlng(tile->header()->base_ll());
+      if (bb.Contains(node_ll)) {
+
+        // If we havent done anything with our bbox yet initialize it
+        if (!min_bb.minpt().IsValid())
+          min_bb = AABB2<PointLL>(node_ll, node_ll);
+
+        // Look at the shape of each edge leaving the node
```

Comment:
do you really need the shape? can you just get the end node and use it to expand the bounding box? The shape will be held in the tile of the end node. Tough I guess if you want to do a loki search near a part of the shape outside both the begin and end node's tile then you would want to include the tile of the shape point (since that tile will have the bins to index into the end node tiles).